R/checkers.R

```diff
@@ -317,3 +323,186 @@ assert_single_value <- function(x, arg = "x",
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#' @param add_err_msg add_err_msg string containing an additional error message,
+#' default is the empty string (`""`)
+#'
+#' @return NULL invisible
+assert_daily_data <- function(dates,
+                              call = rlang::caller_env(),
+                              add_err_msg = "") {
+  # Generate a sequence of dates from the minimum to the
+  # maximum date in the dataset
+  expected_dates <- seq.Date(
+    from = min(dates),
+    to = max(dates),
+    by = "day"
+  )
+
+
+  if (!all(expected_dates %in% dates)) {
+    cli::cli_abort(
+      c(
+        "Vector of dates does not contain dates for each day",
+        add_err_msg
+      ),
+      call = call,
+      class = "wwinference_input_data_error"
+    )
+  }
+  invisible()
+}
+
+#' Assert that the vector of dates spans at least the specified
+#' calibration time
+#'
+#' @param date_vector the vector of dates to check, must be of Date type
+#' @param calibration_time integer indicating the number of days that
+#' the dates must span
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#' @param add_err_msg add_err_msg string containing an additional error message,
+#' default is the empty string (`""`)
+#'
+#' @return NULL invisible
+assert_sufficient_days_of_data <- function(date_vector,
+                                           calibration_time,
+                                           call = rlang::caller_env(),
+                                           add_err_msg = "") {
+  # check that you have sufficient count data for the calibration time
+  calibration_start <- max(date_vector,
+    na.rm = TRUE
+  ) - lubridate::days(calibration_time) + 1
+  check_sufficient_data <- min(date_vector, na.rm = TRUE) <= calibration_start
+  if (!check_sufficient_data) {
+    cli::cli_abort(
+      c(
+        "Insufficient data for specified calibration time"
+      ),
+      call = call,
+      class = "wwinference_specification_error"
+    )
+  }
+  invisible()
+}
+
+#' Assert that two vectors of dates overlap
+#'
+#' @param dates1 the vector of dates to check, must of date type
+#' @param dates2 the vector of dates to compare to, must be of date type
+#' @param max_date the maximum date the testing dates should be
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#' @param add_err_msg add_err_msg string containing an additional error message,
+#' default is the empty string (`""`)
+#'
+#' @return NULL invisible
+assert_overlap_dates <- function(dates1,
+                                 dates2,
+                                 max_date,
+                                 call = rlang::caller_env(),
+                                 add_err_msg = "") {
+  checkmate::assert_date(dates1)
+  checkmate::assert_date(dates2)
+  check_data_overlap <- min(dates1) <= max(dates2) &
+    max(dates1) <= max_date
```

Comment:
Yeah, this is a great flag, this isn't named correctly. What I am really checking is that the second set of dates is within the first date in the first set of dates and the maximum date. Because for example we could have:
```r
hosp_dates<- lubridate::ymd(c("2023-01-01", "2023-01-02"))
ww_dates <- lubridate::ymd(c("2023-01-03", "2023-01-04"))
max_date <- "2023-01-05"
checkmate::assert_date(hosp_dates) # passes
checkmate::assert_date(ww_dates) # passes 
check_data_overlap <- min(hosp_dates) <= max(ww_dates) &
min(ww_dates) <= max(hosp_data) &
    max(ww_dates) <= max_date  &
max(hosp_dates) <=- max_date # TRUE
```

Will revise